### ISSUE TITLE
Linux command line problem

### ISSUE THREAD

**Issue by gwaldbieser** [beiko-lab/MEGASAT#1]
Thanks for this program! I would prefer to use the Linux command line so that I don't have to transfer fastq files to a Windows machine. I am trying to run the test data on Linux but cannot get the command line to accept the inputs, even when using full path information. I have unzipped the test fastq files and renamed the folder to JGPedigree_Fastq (to remove spaces).
-----------------------------------------------------------------------------------------------------
(From within /mnt/data8/MegasatTestData/)
$perl /home/software/MEGASAT/MEGASAT_Genotype.pl   /mnt/data8/MegasatTestData/primer-input.txt   2  5    /mnt/data8/MegasatTestData/Pedigree/JGPedigree_Fastq/       /mnt/data8/MegasatTestData/output

Missing command line arguments!

#######################################################################

Need the directory of primer file as the first command-line argument

Need the maximum number of mismatches as the second command-line argument

Need the minimum depth threshold as the third command-line argument

Need the directory of data set folder that contains all the input sequence read files (fastq or fasta) as the fourth command-line argument

Need the directory to save the output folder as the fifth command-line argument
-----------------------------------------------------------------------------------------------------

I would appreciate if you would you provide more detail on the expected inputs for the command line.
Geoff

**Comment by LuyaoZhan** [comment]
Hello,

I think you are missing the fourth argument that is the number of
processors. This argument is just for doing multiprocessing. If you don't
want to use multiple processors, you can type 1 as the fourth argument in
the command line.

If you have any questions, please let me know.

Best wishes
Catherine

**Comment by gwaldbieser** [comment]
Ah, that makes sense now in light of the parameters entered in the Windows GUI but it wasn't obvious in the error messaging. It's working now.
Thanks!

**Comment by gwaldbieser** [comment]
Initially, I ran the test data which worked as expected. When I tested some of my data only blank genotypes were output. I tracked it down to an error on page 6 of the User's Manual where the column labels for 5' flank and 3' flank were switched. Once I switched those columns in my primer file then everything worked well.

**Comment by LuyaoZhan** [comment]
Thank you for telling me this error in the User's Manual. I didn't notice
before. I will change it to the correct format. Thank you.

Best wishes
Catherine

**Comment by gwaldbieser** [comment]
I have looked at the manuscript online and supplemental material and cannot find details on the scoring rules and how the six scoring ratios are defined. Can you provide that information, please?

**Comment by LuyaoZhan** [comment]
I think we provided the scoring information in the supplemental material.
Anyway, I just found a word file (attached) that explains the scoring
rules. If you have any questions, please let me know.

Thank you!

Best wishes
Catherine

**Comment by gwaldbieser** [comment]
Hi Catherine,
I couldn't find it in the supporting information online (Appendices S1-S4). 
The MS Word file did not attach. Can you post it to the code page?
Thanks.
Geoff

**Comment by LuyaoZhan** [comment]
Hi Geoff,

Could you send me your email address? I can attach that word file in the
email.

Best wishes
Catherine

**Comment by gwaldbieser** [comment]
Geoff.Waldbieser@ars.usda.gov

Thanks!

**Comment by gwaldbieser** [comment]
Hi Catherine,
When using MEGASAT, does the MRA have to be a perfect repeat?

Geoff

**Comment by LuyaoZhan** [comment]
Hi Geoff,

The MRA does not need to be perfect repeat. We allow sequencing errors in
that repeating area.

Thank you.

Best wishes
Catherine

**Comment by gwaldbieser** [comment]
Thanks. I’m trying to track down why some new loci worked and some did not. It must be somewhere in the sequence I selected for the primer file.
Does the FF sequence need to include all sequence between the FP and MRA? Likewise, does the RF sequence need to include everything from the MRA to the RP?

**Comment by LuyaoZhan** [comment]
Yes, the FF will be the sequence between FP and the MRA. Likewise, the RF
will be the sequence between the MRA and RP.

Catherine

**Comment by gwaldbieser** [comment]
The entire sequence? It seems that if I skip some bases between the FP and FF sequences then the genotypes don’t get called, but if I only use ‘X’ for the FF then the genotypes are called. My question is – does the entire 5’ flanking sequence on the input reads need to be accounted for by the FP and FF sequences?

**Comment by LuyaoZhan** [comment]
The reason why the genotypes are called when using "X" for the FF is
because script will assume the locus does not have FF and search for the
other components. In some cases, if you skip some bases between FP and FF,
the genotypes should be called. I would suggest you to include the entire
sequence for FF and RF.

Catherine

**Comment by gwaldbieser** [comment]
Hi Catherine,
I am wondering why the sometimes the algorithm identifies two major allele counts but calls only one allele in the genotype. This happens in a locus that has a very large number of alleles. Here are examples where I used the default basecall settings in the primer file.
Name    left_oligo      Rev_Comp_right_oligo    5Flank  3Flank  Repeat
c19-02  CCTACGATTAGTGGACAACCCGCTCTACCACCTGAGCCACAGCTG   TAGTGAAAGCCCAGTTCGGTG   X       ACCCACAGCAGCAGT GATA

Here are results in the length distribution file ( I have removed columns with zero reads).
Microsatellite

39

85

111

127

133

136

137

141

157

169

171

175

183

187

195

199

217

221

225

229

233

sum

scores

c19-02

2

1

1

1

7

1

227

1

1

3

1

8

1

8

1

1

1

24

454

4

3

751

225 225

Microsatellite

77

111

133

137

141

169

175

177

187

191

217

221

224

225

229

253

sum

scores


c19-02

1

1

14

297

1

2

2

1

11

1

1

16

1

571

5

1

926

225 225




Why were these samples not be genotyped as 137/225?

Geoff

**Comment by gwaldbieser** [comment]
I think I see it now. A1>A2 and A2/A1 is less than R4 (default value=0.6)?


From: Waldbieser, Geoff
Sent: Friday, June 09, 2017 11:21 AM
To: 'beiko-lab/MEGASAT' <reply@reply.github.com>
Subject: RE: [beiko-lab/MEGASAT] Linux command line problem (#1)

Hi Catherine,
I am wondering why the sometimes the algorithm identifies two major allele counts but calls only one allele in the genotype. This happens in a locus that has a very large number of alleles. Here are examples where I used the default basecall settings in the primer file.
Name    left_oligo      Rev_Comp_right_oligo    5Flank  3Flank  Repeat
c19-02  CCTACGATTAGTGGACAACCCGCTCTACCACCTGAGCCACAGCTG   TAGTGAAAGCCCAGTTCGGTG   X       ACCCACAGCAGCAGT GATA

Here are results in the length distribution file ( I have removed columns with zero reads).
Microsatellite

39

85

111

127

133

136

137

141

157

169

171

175

183

187

195

199

217

221

225

229

233

sum

scores

c19-02

2

1

1

1

7

1

227

1

1

3

1

8

1

8

1

1

1

24

454

4

3

751

225 225

Microsatellite

77

111

133

137

141

169

175

177

187

191

217

221

224

225

229

253

sum

scores



c19-02

1

1

14

297

1

2

2

1

11

1

1

16

1

571

5

1

926

225 225





Why were these samples not be genotyped as 137/225?

Geoff
